### ISSUE TITLE
Branches no longer identify related Pull Request

### ISSUE THREAD

**Issue by danpalmer** [desktop/desktop#11617]
### Describe the bug

When on a branch with a Pull Request, Desktop used to show information relating to the Pull Request, but now does not.

### Version & OS

macOS Catalina
Desktop: 2.6.3

### Steps to reproduce the behavior

1. Create branch with content.
2. Create pull request on GitHub for that branch.
3. Refresh GitHub Desktop
4. See that the branch does not show PR related info.

### Expected behavior

Desktop used to show the PR status in the branch selector, present a button to open the pull request, and use the Cmd+R shortcut to open the PR.

### Actual behavior

Recently this has stopped working, a branch with a PR open does not show the check status, is showing a button to "Create Pull Request", and Cmd+R takes me to the comparison page, not the PR.

The Pull Request is shown in the PR selector menu, and clicking on it checks out `pr/ID`.

### Screenshots

Not attached as a branch with a PR just looks identical to one without a PR.

### Logs

Not attached as I'm hoping they aren't necessary and don't have time to censor them. Let me know if needed.

**Comment by sergiou87** [comment]
Hi @danpalmer! 👋 

I'm sorry about this weird behavior you're experiencing. I just tried it with branches corresponding to open PRs and everything works as expected, I get the PR indicator and Cmd+R shows the existing PR:

![image](https://user-images.githubusercontent.com/1083228/108345281-5ce32300-71de-11eb-8dd1-f997cbcea4e5.png)
![image](https://user-images.githubusercontent.com/1083228/108345313-64a2c780-71de-11eb-8957-b437ee5c4914.png)

![image](https://user-images.githubusercontent.com/1083228/108345358-75533d80-71de-11eb-858e-ac416e8ebd65.png)

Is this something that happens to you in all repos and all branches? 🤔

**Comment by danpalmer** [comment]
@sergiou87 Upon further inspection it appears to happen inconsistently across repos (some work, some don't), and consistently within repos (if a repo works it always works, etc).

I've checked the repo configuration and the remotes for a repo that works and a repo that doesn't are specified in the same format (so this isn't an HTTPS vs git issue). I've also checked my gitconfig and I don't have anything special set up for either repo. Also tried quitting and restarting to see if that helps, but no change.
